core/kernel/user_ta.c

```diff
@@ -486,6 +486,22 @@ TEE_Result tee_ta_init_user_ta_session(const TEE_UUID *uuid,
 #endif
 
 	mutex_lock(&tee_ta_mutex);
+
+	/*
+	 * Before updating the context list check again if a context
+	 * already exists in the list for a single instance TA, maybe
+	 * created recently enough so that it was not found at entry
+	 * in tee_ta_init_session().
+	 */
+	res = tee_ta_init_session_with_context(s, uuid);
+	if (res != TEE_ERROR_ITEM_NOT_FOUND) {
+		vm_info_final(&utc->uctx);
```

Comment:
I'd prefer checking if the context is available before allocating the `utc`. That means holding the mutex during the call to calloc(), but that should be OK.